src/yoroi-wallets/cardano/ShelleyWallet.ts

```diff
@@ -1178,6 +1181,8 @@ export class ShelleyWallet implements WalletInterface {
 
     // later only if utxos have changed
     await this.transactionCache.doSync(this.getAddressesInBlocks(), this.getBackendConfig())
+
+    this.updateLastGeneratedAddressIndex()
```

Comment:
Fix. The index update is done after transaction cache and the external chain are synced.

Reply:
Question: maybe the issue is that `this.getAddressesIBlocks()` was supposed to be `async` ?

Reply:
As discussed, the index depends on the transaction cache. Thanks for the question!